src/hotspot/share/gc/z/zBarrierSet.cpp

```diff
@@ -42,7 +42,7 @@ class ZBarrierSetC2;
 
 static BarrierSetNMethod* make_barrier_set_nmethod() {
   // NMethod barriers are only used when class unloading is enabled
-  if (!ClassUnloading) {
+  if (!ZHeap::heap()->should_unload_class()) {
```

Comment:
Here should be 
(ClassUnloading && ZUnloadClassesFrequency != 0)

Reply:
`ZUnloadClassesFrequency != 0` is in `ZHeap::should_unload_class()`